lib/src/rules/hash_and_equals.dart

```diff
@@ -7,12 +7,15 @@ import 'package:analyzer/dart/ast/visitor.dart';
 import 'package:linter/src/analyzer.dart';
 import 'package:linter/src/ast.dart';
 
-const desc = r'Always override `hashCode` if overriding `==`.';
+const _desc = r'Always override `hashCode` if overriding `==`.';
 
-const details = r'''
+const _details = r'''
 
 **DO** override `hashCode` if overriding `==`.
 
+Every object in Dart should have a `hashCode`. This is necessary so that
+every Dart object can have its `hashCode` used as a key in a map.
```

Comment:
Every object has a hashCode. The reason it's important for `==` and `hashCode` to be consistent is so that a common implementation of hash maps will work correctly, and the reason to override both together is to maintain that consistency.